ledger/participant-integration-api/src/main/scala/platform/store/backend/oracle/OracleStorageBackend.scala

```diff
@@ -0,0 +1,520 @@
+      transactionId = transactionId,
+      requestingParty = requestingParty,
+      partyArrayContext = partyArrayContext,
+      witnessesWhereClause =
+        arrayIntersectionWhereClause("tree_event_witnesses", Set(requestingParty)),
+      submittersInPartiesClause = arrayIntersectionWhereClause("submitters", Set(requestingParty)),
+      columnEqualityBoolean = columnEqualityBoolean("event_kind", "20"),
+    )(connection)
+
+  def transactionTreeMultiParty(
+      transactionId: TransactionId,
+      requestingParties: Set[Ref.Party],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]] =
+    TemplatedStorageBackend.transactionTreeMultiParty(
+      transactionId = transactionId,
+      witnessesWhereClause =
+        arrayIntersectionWhereClause("tree_event_witnesses", requestingParties),
+      submittersInPartiesClause = arrayIntersectionWhereClause("submitters", requestingParties),
+      filteredWitnessesClause = arrayIntersectionValues("tree_event_witnesses", requestingParties),
+      columnEqualityBoolean = columnEqualityBoolean("event_kind", "20"),
+    )(connection)
+
+  def transactionTreeEventsSingleParty(
+      startExclusive: Long,
+      endInclusive: Long,
+      requestingParty: Ref.Party,
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]] =
+    TemplatedStorageBackend.transactionTreeEventsSingleParty(
+      startExclusive = startExclusive,
+      endInclusive = endInclusive,
+      requestingParty = requestingParty,
+      partyArrayContext = partyArrayContext,
+      witnessesWhereClause =
+        arrayIntersectionWhereClause("tree_event_witnesses", Set(requestingParty)),
+      limitExpr = limitClause(limit),
+      fetchSizeHint = fetchSizeHint,
+      submittersInPartiesClause = arrayIntersectionWhereClause("submitters", Set(requestingParty)),
+      columnEqualityBoolean = columnEqualityBoolean("event_kind", "20"),
+    )(connection)
+
+  def transactionTreeEventsMultiParty(
+      startExclusive: Long,
+      endInclusive: Long,
+      requestingParties: Set[Ref.Party],
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]] =
+    TemplatedStorageBackend.transactionTreeEventsMultiParty(
+      startExclusive = startExclusive,
+      endInclusive = endInclusive,
+      witnessesWhereClause =
+        arrayIntersectionWhereClause("tree_event_witnesses", requestingParties),
+      filteredWitnessesClause = arrayIntersectionValues("tree_event_witnesses", requestingParties),
+      submittersInPartiesClause = arrayIntersectionWhereClause("submitters", requestingParties),
+      limitExpr = limitClause(limit),
+      fetchSizeHint = fetchSizeHint,
+      columnEqualityBoolean = columnEqualityBoolean("event_kind", "20"),
+    )(connection)
+
+  // TODO FIXME: confirm this works for oracle
```

Comment:
just checking: do you want something to do about this TODO, or shall we keep it and we take care about this TODO in cleanup/oracle perf tuning epic?

Reply:
removed